### PR TITLE
Additional CPS Use Cases

### PR DESCRIPTION
**Pull request by TobyConsidine**
1 for Reporting & Monitoring
1 for Tripwires

### CODE REVIEW

File: Operational Technology (IOT)/Reporting & Monitoring [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+Cybersecurity monitoring if such systems may include monitoring the “other”
+systems, the services they provide, or the net effect of all of the systems
+working together.
+
+This leads to a requirement for what I call a reporting function. A report may
+be immediate (what is going on now), delayed (report back at 3:00 or in two
+hours), triggered (report if this ever happens). It may be rate driven (report
+if this happens more then 5 times an hour or 5 times a minute). It may track
+longer term monitoring, and the monitoring could be event based (count each time
+you see this) or episodic (right down the flow rate every five minutes). The
```
Comment (dlemire60) [inline Operational Technology (IOT)/Reporting & Monitoring:28]: line 28:  "right" should be "write"

File: Operational Technology (IOT)/Reporting & Monitoring [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+finite in duration, but that may be different on a wet road or an icy one. In
+any case, stopping faster than the car behind can do may also result in
+cataclysmic failure.
+
+On a power transmission grid, blocking the software that operates a substation
+with a DDOS Attack may be indistinguishable in effect from simply shooting a
+transformer with a rifle. CPS cybersecurity must always consider the blending of
+cyber and physical.
+
+A CPS system, whether a microgrid or factory floor or a wet utility or a swarm
```
Comment (dlemire60) [inline Operational Technology (IOT)/Reporting & Monitoring:17]: "wet utility"?  Seems odd wording, did you mean "water utility"?

File: Operational Technology (IOT)/Reporting & Monitoring [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+of drones, likely consists of multiple autonomous or semiautonomous systems.
+Cybersecurity monitoring if such systems may include monitoring the “other”
```
Comment (dlemire60) [inline Operational Technology (IOT)/Reporting & Monitoring:19]: In "... monitoring if such systems ...", "if" should be "of"

File: Operational Technology (IOT)/Reporting & Monitoring [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+knowing its baseline, and being able to report variance from that baseline.
+
+EI Reports are simple Streams with some metadata identifying the report and a
+collection of Intervals containing the Payloads for each [measurement]. Reports
+can be of the past, the present, or the future. A Report appears as a series of
+[measurements] in the past. A Snap is a Report made as of a single moment. A
+Projection is in the same form as a report, but it includes projections of what
+will be in the future, including a confidence level in the payload.
+
+EI Reported values be Actual, Summed (across several sources), Derived based on
```
Comment (dlemire60) [inline Operational Technology (IOT)/Reporting & Monitoring:51]: In "EI Reported values be Actual, Summed ...", should it be "EI Reported values **_can_** be Actual, Summed ..."?

**Comment by dlemire60** [comment]
Regarding Reporting & Monitoring, it appears that there are existing specifications that define formats and, if I understand correctly, conditions regarding reporting. It seems to me there's a missing link needed to define an OC2 use case, which is the role OC2 should play in C2 for the reporting functions in those standards. For example OC2 could:

- Use a `set` action to configure reporting parameters. 
- Use 'start', `stop`, `restart` actions to manage reporting operations
- Use a `query` action to retrieve reports, somewhat similar to retrieving an SBoM.

Further thought would lead to others. I do think there are aspects of this that clearly go beyond just what a transfer protocol supports, and should be addressed as language and/or AP matters.

File: Operational Technology (IOT)/Tripwires [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+each purpose-built for the CPS in question.
+
+-   Current work in power distribution systems suggest that synchrophasor data
+    collection and analysis could be used as a tripwire.
+
+-   Transactive Resource Management (TRM) systems use internal markets to
+    balance supply and demand of a commodity resource over time—meaning an
+    internal market is the internal coordination mechanism. Security appliances,
+    borrowed form the world of finance, use AI to detect market gaming and
+    market health. Such
```
Comment (dlemire60) [inline Operational Technology (IOT)/Tripwires:25]: Dangling "Such" .  Delete?

**Comment by dlemire60** [comment]
Regarding Tripwires, what's described here sounds like what are often called "Alerts", created by a local (to the sensor) analytic function. So what distinguishes Tripwires from Alerts, and is the difference is meaningful to OpenC2?  In addition, I think there are similar questions as for Reporting & Monitoring regarding what the role of OpenC2 is here. I think alerting is a legitimate feature to include in OpenC2, we just need to build some more detail into the use case.

**Comment by sparrell** [comment]
Please change directory name from Operational Technology to UNC to reflect norms of TC that use cases are for a member organization.

File: Operational Technology (IOT)/Reporting & Monitoring (hunk 2 above)
Comment (TobyConsidine) [reply Operational Technology (IOT)/Reporting & Monitoring]: Wet Utility is a term that applies to Water, Sewer, and Storm Drains. The term is considered a standard construction contractor classification. ("Once the Wet Utility contractor gets out, we can pour the slabs.") In operation, it described the general case of those things that need to flow downhill or to be pumped. As such, it is a common term used in Facilities Management.

File: Operational Technology (IOT)/Reporting & Monitoring (hunk 3 above)
Comment (TobyConsidine) [reply Operational Technology (IOT)/Reporting & Monitoring]: Agrred

File: Operational Technology (IOT)/Tripwires (hunk 5 above)
Comment (TobyConsidine) [reply Operational Technology (IOT)/Tripwires]: Done

File: Operational Technology (IOT)/Reporting & Monitoring (hunk 4 above)
Comment (TobyConsidine) [reply Operational Technology (IOT)/Reporting & Monitoring]: Agreed

**Comment by TobyConsidine** [comment]
Several comments would be useful if an actual spec were to be written, but not necessarily in the Use Case. The Use Case posits the issue, and suggests a couple approaches the Specification editors, as guided by the TC and comments may choose to use. 
For example, an Alert is something that arrives as part of the normal functioning of, say a gateway router. The router's purpose is not the Alert, but it is a natural add-on to its primary role of moving traffic. What I am trying to establish here is a class of devices which may have no purpose other than to provide security alerts-hence the term tripwire. It may well be that what is Alerted is not configurable by the OpenC2 client other than to indicate that it wants to "subscribe". Consider an OpenC2 in a data center setting up many firewall-based alerts AS WELL AS subscribing to the Fire Alarm.
I am OK with the intended specification deciding to use the  term Tripwire, or some other term, or even not at all. A priori, though I think there is a difference between a thing that is purpose built for Alerting, and which may not be configurable, and a system built for general purpose for which an Alert can be configured.
The Physical in a CPS may be outside the ken of the cybersecurity agent or officer.

**Comment by dlemire60** [comment]
I like the updates @TobyConsidine has made. These Use Cases look good to me.

**Pull request merged**